### ISSUE TITLE
Gallery image: fix the elements order

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#14931]
Noticed while testing https://github.com/WordPress/gutenberg/pull/14822

In the gallery images, the order of the elements should be improved. Currently, the order is:
- "Remove Image" button (see `block-library-gallery-item__inline-menu`)
- the image (which is focusable)
- the caption field

With this order, there are at least 2 issues:
- when using a keyboard, the "Remove Image" button gets skipped because it appears _after_ the image is focused
- even if it wasn't skipped, the "Remove Image" button doesn't give any context on _which_ image it will remove because it's _before_ any relevant information related to the image

I'd propose to:
- move the "Remove Image" button after the image: shouldn't be a big problem as it's absolutely positioned
- make sure the button appears both when tabbing forward and when tabbing backwards

**Comment by nosolosw** [comment]
Closing this in favor of pre-existing https://github.com/WordPress/gutenberg/issues/14814 I went ahead and added your feedback there.

**Comment by nosolosw** [comment]
Also: there is a PR at https://github.com/WordPress/gutenberg/issues/14814 to solve that issue. Looking for some feedback there :)

**Comment by afercia** [comment]
Thanks @nosolosw ! Was searching through the "Accessibility" label and missed it. Added the label there.
